UnitsNet.Serialization.JsonNet.Tests/UnitsNetBaseJsonConverterTest.cs

```diff
@@ -0,0 +1,216 @@
+                    sut,
+                    new DataTableConverter()
+                },
+                ContractResolver = new CamelCasePropertyNamesContractResolver()
+            });
+
+            var result = sut.Test_CreateLocalSerializer(serializer);
+
+            Assert.Equal(TypeNameHandling.Arrays, result.TypeNameHandling);
+            Assert.Equal(2, result.Converters.Count);
+            Assert.Collection(result.Converters,
+                (converter) => Assert.IsType<BinaryConverter>(converter),
+                (converter) => Assert.IsType<DataTableConverter>(converter));
+            Assert.IsType<CamelCasePropertyNamesContractResolver>(result.ContractResolver);
+        }
+
+        [Fact]
+        public void UnitsNetBaseJsonConverter_ReadValueUnit_work_as_expected()
+        {
+            var token = new JObject();
+
+            token.Add("Unit", unitName);
+            token.Add("Value", unitValue);
+
+            var result = sut.Test_ReadValueUnit(token);
+
+            Assert.NotNull(result);
+            Assert.Equal(unitName, result?.Unit);
+            Assert.Equal(unitValue, result?.Value);
+        }
+
+        [Fact]
+        public void UnitsNetBaseJsonConverter_ReadValueUnit_works_with_empty_token()
+        {
+            var token = new JObject();
+
+            var result = sut.Test_ReadValueUnit(token);
+
+            Assert.Null(result);
+        }
+
+        [Theory]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        public void UnitsNetBaseJsonConverter_ReadValueUnit_returns_null_when_unit_or_value_is_missing(bool withUnit, bool withValue)
+        {
+            var token = new JObject();
+
+            if (withUnit)
+            {
+                token.Add("Unit", unitName);
+            }
+
+            if (withValue)
+            {
+                token.Add("Value", unitValue);
+            }
+
+            var result = sut.Test_ReadValueUnit(token);
+
+            Assert.Null(result);
+        }
+
+        [Theory]
+        [InlineData("Unit", "Value")]
+        [InlineData("unit", "Value")]
+        [InlineData("Unit", "value")]
+        [InlineData("unit", "value")]
+        [InlineData("unIT", "vAlUe")]
+        public void UnitsNetBaseJsonConverter_ReadValueUnit_works_case_insensitive(string unitPropertyName, string valuePropertyName)
+        {
+            var token = new JObject();
+
+            token.Add(unitPropertyName, unitName);
+            token.Add(valuePropertyName, unitValue);
+
+            var result = sut.Test_ReadValueUnit(token);
+
+            Assert.NotNull(result);
+            Assert.Equal(unitName, result?.Unit);
+            Assert.Equal(unitValue, result?.Value);
+        }
+
+        /// <summary>
+        /// Dummy converter, used to access protected methods on abstract UnitsNetBaseJsonConverter{T}
+        /// </summary>
+        private class TestConverter : UnitsNetBaseJsonConverter<string>
```

Comment:
Hm, why do we need to `access protected methods on abstract UnitsNetBaseJsonConverter`?
I love tests, but is it really useful to test the implementation details of an abstract class?
Isn't it enough to test the public members of the derived types?

I mean, in the end, we really just care about it being serialized and deserialized correctly.

Reply:
what you are referring to are the integration tests imo. 
These are actual unit tests.
This way, if anything changes inside the base converter we'll notice straight away.
Otherwise, i'd have to test the shared functions on either both or one of the converters which makes it unclear.
Not explicitly testing the functionality is a no-no

Reply:
Alright, we'll keep it